portfolio/src/main/java/com/google/sps/servlets/BookServlet.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Book;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Returns book titles and reviews as a JSON array, e.g. [{"title": Othello, "reviews": {"Nice",
+ * "Bad"}}]
+ */
+@WebServlet("/book-data")
+public class BookServlet extends HttpServlet {
+  private Collection<Book> books;
+
+  @Override
+  public void init() throws ServletException {
+    books = new ArrayList<>();
+    readBooks();
```

Comment:
(feel free to do this in separate PR)
`books = readBooks();` is a preferable way to achieve this by making it a pure function, where readBooks can be invoked multiple times, without any side effects (changes to instance variables).